File: 2609-find-the-longest-balanced-substring-of-a-binary-string/2609-find-the-longest-balanced-substring-of-a-binary-string.cpp
class Solution {
public:
    int findTheLongestBalancedSubstring(string s) {
        int result=0;
        int zeros=0;
        int ones=0;
        int temp=0;
        
        for(int i=0; i<s.length();i++)
        {
            if(s[i]=='0'){ zeros++; }
        
        else
        {
            ones++;
        }
        temp++;
        result=max(result, 2*min(ones, zeros));
        if(i+1<s.length() && s[i]=='1' && s[i+1]=='0')
        {
            zeros=ones=temp=0;
        }
        }
        return result;
    }
};

File: 2609-find-the-longest-balanced-substring-of-a-binary-string/2609-find-the-longest-balanced-substring-of-a-binary-string.java
class Solution {
    public int findTheLongestBalancedSubstring(String s) {
        int result=0;
        String temp="01";
        
        while(temp.length()<=s.length())
        {
            if(s.contains(temp))
            {
                result=temp.length();
            }
            temp="0"+temp+"1";
        }
        return result;
    }
}